src/Symfony/Bundle/WebProfilerBundle/EventListener/TurboDriveCspListener.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\WebProfilerBundle\EventListener;
+
+use Symfony\Component\DependencyInjection\Attribute\When;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\HttpKernel\KernelInterface;
+
+/**
+ * Show the Web Debug Toolbar (WDT) when navigating via Turbo Drive and when a strict Content-Security-Policy is set.
+ * This is done by reusing WDT's nonces.
+ * This service is registered only when TurboBundle is installed.
+ *
+ * @author David Petrásek <davidpetrasek@hotmail.cz>
+ */
+#[When(env: 'dev')]
+class TurboDriveCspListener implements EventSubscriberInterface
+{
+    public function __construct(
+        readonly private KernelInterface $kernel,
```

Comment:
Injecting the whole kernel isn't necessary. The debug flag alone should be enough.

```php
#[Autowire(param: 'kernel.debug')]
private readonly bool $isDebug,
```

Reply:
you don't even need this Autowire attribute as we configure our services explicitly instead of relying on autowiring.

Reply:
Indeed, it is explicitly set in the service definition.